pkgs/applications/virtualization/gvisor/default.nix

```diff
@@ -0,0 +1,176 @@
+{
+  stdenv
+, pkgs
+, fetchFromGitHub
+, cacert
+, symlinks
+, writeScript
+
+, coreutils
+, bash
+, bazel
+, git
+, go
+, python
+}:
+
+let
+
+  # gvisor doesn't have any releases yet
+  version = "2018-11-10";
+
+  # From the Bazel docs:
+  #   Beneath the outputUserRoot directory, we also create an outputBase
+  #   directory whose name is the MD5 hash of the path name of the workspace
+  #   directory.
+  #
+  # The default `fetchzip` arguments will unpack our file as the name `source`,
+  # and the Nix sandbox builds things in the `/build` directory. Thus, we can
+  # predict this hash and calculate it here:
+  outputBaseHash = builtins.hashString "md5" "/build/source";
+
+  # Source from GitHub with additional patches.
+  patchedSource = fetchFromGitHub {
+    owner = "google";
+    repo  = "gvisor";
+    rev   = "d97ccfa346d23d99dcbe634a10fa5d81b089100d";
+
+    # NOTE: this is the output of the whole fixed-output derivation, so
+    # `nix-prefetch-git` won't work to obtain this. The easiest way is to just
+    # change it and see what breaks :)
+    sha256 = "1bcnq7kazbf6l5j0g82x2lvg1nbp7z70klk139dxi0jkw0j8dh3r";
```

Comment:
This is not the expected hash. Maybe, you forgot to update it when you changed line 63.

Reply:
Ahhhhhhhh, I know what this is.  The fixed-output derivation has the paths to `bash` and `coreutils` as part of the derivation, so any change to those results in a new hash here too.  That's annoying :unamused: 

I think I can fix that by changing the `bash` calls to `/bin/sh`, and just dropping the test fixes since we're not running them anyway.

Reply:
If possible, it would be better to patch shebangs in the patch phase of the `bazelDependencies` derivation.